File: src/ehs/NvmMemCtrl.py
from m5.objects.MemCtrl import MemCtrl


class NvmMemCtrl(MemCtrl):
    """A MemCtrl that lets the power model read the traffic it has served.

    Main-memory traffic is the largest energy term on this machine

    MemCtrl already counts the bytes and only keeps them protected; this adds
    no parameters and no behaviour, just the two accessors.
    """

    type = "NvmMemCtrl"
    cxx_class = "gem5::memory::NvmMemCtrl"
    cxx_header = "ehs/nvm_mem_ctrl.hh"

File: src/ehs/nvm_mem_ctrl.cc
#include "ehs/nvm_mem_ctrl.hh"

namespace gem5
{
namespace memory
{

NvmMemCtrl::NvmMemCtrl(const Params &p)
  : MemCtrl(p)
{
}

uint64_t
NvmMemCtrl::bytesRead() const
{
    return uint64_t(stats.bytesReadSys.value());
}

uint64_t
NvmMemCtrl::bytesWritten() const
{
    return uint64_t(stats.bytesWrittenSys.value());
}

} // namespace memory
} // namespace gem5
